### DISCUSSION TITLE
TemporaryUploadedFile getPath() method returns tmp path

### DISCUSSION THREAD

**Discussion by daylightstudio** [livewire/livewire#3064] [7. Bugs (archived)]
I'm using Spatie's Media Library package and that package uses the `getPath()` method:
https://github.com/spatie/laravel-medialibrary/blob/227b47bd001df05a7f3fdc7c7e5d801def61be27/src/MediaCollections/FileAdder.php#L104

However, the `getPath()` method is returning a path to a file that doesn't exist and is causing upload errors:
```
/private/var/tmp/qObFvXUTXerwxsHFN2aHjrmjdNTRdG-metabGlsbGFyZF9ieWVfYnllLmdpZg==-.gif
```

I noticed that the `TemporaryUploadedFile` class isn't overwriting that method like the others to create the correct path and it seems like it should?
https://github.com/livewire/livewire/blob/b647ac63378ea02bdac975f70a4e57aea84bda0b/src/TemporaryUploadedFile.php#L21

Suggestion would be to add the something like the following:
```
public function getPath()
{
    return dirname($this->storage->path($this->path));
}
```

**Comment by tahirafridi** [comment]
I'm facing the same problem. Medialibrary can not read the temp file from livewire, as livewire made it protected and hashed so it can not be accessible by other libraries. Anyone here can provide work around?

**Comment by tahirafridi** [comment]
> I'm using Spatie's Media Library package and that package uses the `getPath()` method:
> https://github.com/spatie/laravel-medialibrary/blob/227b47bd001df05a7f3fdc7c7e5d801def61be27/src/MediaCollections/FileAdder.php#L104
> 
> However, the `getPath()` method is returning a path to a file that doesn't exist and is causing upload errors:
> 
> ```
> /private/var/tmp/qObFvXUTXerwxsHFN2aHjrmjdNTRdG-metabGlsbGFyZF9ieWVfYnllLmdpZg==-.gif
> ```
> 
> I noticed that the `TemporaryUploadedFile` class isn't overwriting that method like the others to create the correct path and it seems like it should?
> https://github.com/livewire/livewire/blob/b647ac63378ea02bdac975f70a4e57aea84bda0b/src/TemporaryUploadedFile.php#L21
> 
> Suggestion would be to add the something like the following:
> 
> ```
> public function getPath()
> {
>     return dirname($this->storage->path($this->path));
> }
> ```

I make it work
Because using livewire, file already uploaded to livewire-tmp directory, so here I used
`
$row->addMedia($this->image->path())->toMediaCollection('original');
`

Notice addMedia() method, and also the path() method from livewire. Before I was trying to addMediaFromRequest() and it was failing, because the $this->image has uploaded file not the requested file. I hope it will help you some how!

**Comment by daylightstudio** [comment]
Thanks @tahirafridi, that did the trick for me! Much appreciated.

Seems like that method should still probably be updated unless it was excluded for some reason.

**Comment by tahirafridi** [comment]
I'm glad that it works for you @daylightstudio

**Reply by njoguamos** [reply]
Worked for me too. Thanks.

**Comment by amiranagram** [comment]
This is fixed in https://github.com/livewire/livewire/pull/4001 @daylightstudio.

**Comment by MCKLtech** [comment]
I came across this discussion and PR as I'm encountering a similar error with S3 . S3 is used for both the Laravel disk and for Livewire tmp directory.

Using FilePond, after I upload the files and attempt to save them, I'm getting this error:

`File `livewire-tmp/ZUkXKIXwRVcEmaNMzFv9oqGjrujL3Y-metaVGhpbmtpZmljX09BdXRoXzIuMC5wZGY=-.pdf` does not exist`

However, that file does exist on S3 in the livewire-tmp folder.

The code for saving the files once FilePond has handled this is this:

 ```

public $files = [];

public function uploadMedia()
    {
        foreach ($this->files as $file) {

            /**
             * Does not work
             */
            $this->funnel->addMedia($file)->toMediaCollection();
            
            /**
             * Does not work
             */
            $this->funnel->addMedia($file->path())->toMediaCollection();
            
            //Works but the file name is lost i.e. It maintains the file name of the temp upload from Livewire
            $this->funnel->addMediaFromDisk($file->path(), 's3')->toMediaCollection();
        }

        $this->dispatchBrowserEvent('pondReset');
        $this->emit('uploaded');
        $this->reset(['files']);
    }
```

`<x-filepond wire:model="files" multiple/>`
```

<div
    wire:ignore
    x-data
    x-init="
        FilePond.setOptions({
            allowMultiple: {{ isset($attributes['multiple']) ? 'true' : 'false' }},
            server: {
                process: (fieldName, file, metadata, load, error, progress, abort, transfer, options) => {
                    @this.upload('{{ $attributes['wire:model'] }}', file, load, error, progress)
                },
                revert: (filename, load) => {
                    @this.removeUpload('{{ $attributes['wire:model'] }}', filename, load)
                },
            },
        });
        var filePondObj = FilePond.create($refs.input);
        window.addEventListener('pondReset', e => {
            filePondObj.removeFiles();
        });
    "
>
    <input type="file" x-ref="input">
</div>

@push('header-scripts')
    <link href="https://unpkg.com/filepond/dist/filepond.css" rel="stylesheet">
@endpush
@push('footer-scripts')
    <script src="https://unpkg.com/filepond/dist/filepond.js"></script>
@endpush



```

**Reply by Enaah** [reply]
@MCKLtech did you find a solution? 

I am have a similar problem right now (File exists (preview does work aswell) but adding it to the medialibrary collection does not work.
